group4/81966_Georgi_Gerginov/Set/comp_implementation/set/set.cpp

```diff
@@ -0,0 +1,148 @@
+#include "set.h"
+
+Set::Node::Node(Comparable* _data, Node* _next = nullptr, Node* _prev = nullptr) : data{_data}, next{_next}, prev{_prev} {}
+
+void Set::del() {
+    while(m_start -> next) {
+        m_start = m_start -> next;
+        delete m_start -> prev;
+    }
+
+    delete m_start;
+}
+
+bool Set::is_unique(Comparable* _data) const {
+    Node* curr{m_start};
+
+    while(curr) {
+        if(curr -> data -> compareTo(_data) == 0) { 
+            return false;
+        }
+        curr = curr -> next;
+    }
+
+    return true;
+}
+
+Set::Set() : m_length{0}, m_start{nullptr}, m_end{nullptr} {}
+
+Set::Set(const std::initializer_list<Comparable*>& init_list) : Set{} {
+	for(Comparable* curr : init_list) {
+		push(curr);
+	}
+}
+
+Set::~Set() {
+    del();
+}
+
+void Set::push_front(Comparable* new_data) {
```

Comment:
This is ok for unordered set but if the set is ordered you can have only method add because you are not choosing the position.